From 409a59343e0ea731265b7bd5025b4c4820c58c83 Mon Sep 17 00:00:00 2001
From: "Kothapeta, BikshapathiX" <bikshapathix.kothapeta@intel.com>
Date: Tue, 7 Feb 2023 08:01:15 +0530
Subject: [PATCH] Update-mixins for codec xml

update media_codecs_intel_c2_video.xml as per below PR
https://github.com/projectceladon/MediaSDK_C2/pull/70

Tracked-On: OAM-105460
Signed-off-by: Kothapeta, BikshapathiX <bikshapathix.kothapeta@intel.com>

diff --git a/caas/extra_files/codec2/media_codecs_intel_c2_video.xml b/caas/extra_files/codec2/media_codecs_intel_c2_video.xml
index eed1381..45b1242 100644
--- a/caas/extra_files/codec2/media_codecs_intel_c2_video.xml
+++ b/caas/extra_files/codec2/media_codecs_intel_c2_video.xml
@@ -18,45 +18,51 @@
 from AOSP frameworks/av/media/libstagefright/data/media_codecs_google_c2_video.xml
 and updated to vendor media codecs.
 -->
+
+<!--
+    Max block-count = maxHeight / block-size(height) * maxWidth / block-size(W)
+    Max blocks-per-second = Max block-count * frames-per-second
+-->
+
 <Included>
     <Decoders>
-        <MediaCodec name="c2.intel.avc.decoder" type="video/avc">
+	<MediaCodec name="c2.intel.avc.decoder" type="video/avc">
             <!-- profiles and levels:  ProfileHigh : Level52 -->
             <Limit name="size" min="64x64" max="4096x4096" />
             <Limit name="alignment" value="2x2" />
             <Limit name="block-size" value="16x16" />
-            <Limit name="block-count" range="1-32768" /> <!-- max 4096x2048 equivalent -->
+            <Limit name="block-count" range="1-65536" /> <!-- max 4096x4096 equivalent -->
             <Limit name="blocks-per-second" range="1-1966080" />
-            <Limit name="bitrate" range="1-48000000" />
+            <Limit name="bitrate" range="1-40000000" />
             <Limit name="performance-point-3840x2160" value="30" />
             <Feature name="adaptive-playback" />
-        </MediaCodec>
+	</MediaCodec>
 
-        <MediaCodec name="c2.intel.hevc.decoder" type="video/hevc">
+	<MediaCodec name="c2.intel.hevc.decoder" type="video/hevc">
             <!-- profiles and levels:  ProfileMain : MainTierLevel51 -->
             <Limit name="size" min="64x64" max="8192x8192" />
             <Limit name="alignment" value="2x2" />
             <Limit name="block-size" value="8x8" />
-            <Limit name="block-count" range="1-196608" /> <!-- max 4096x3072 -->
-            <Limit name="blocks-per-second" range="1-2000000" />
-            <Limit name="bitrate" range="1-10000000" />
+            <Limit name="block-count" range="1-1048576" /> <!-- max 8192x8192 -->
+            <Limit name="blocks-per-second" range="1-31457280" />
+            <Limit name="bitrate" range="1-40000000" />
             <Limit name="performance-point-3840x2160" value="30" />
             <Feature name="adaptive-playback" />
-        </MediaCodec>
+	</MediaCodec>
 
-        <MediaCodec name="c2.intel.vp9.decoder" type="video/x-vnd.on2.vp9">
+	<MediaCodec name="c2.intel.vp9.decoder" type="video/x-vnd.on2.vp9">
             <Limit name="size" min="64x64" max="8192x8192" />
             <Limit name="alignment" value="2x2" />
             <Limit name="block-size" value="16x16" />
-            <Limit name="block-count" range="1-16384" />
-            <Limit name="blocks-per-second" range="1-500000" />
+            <Limit name="block-count" range="1-262144" />
+            <Limit name="blocks-per-second" range="1-7864320" />
             <Limit name="bitrate" range="1-40000000" />
             <Limit name="performance-point-3840x2160" value="30" />
-            <Feature name="adaptive-playback" />
-        </MediaCodec>
+	    <Feature name="adaptive-playback" />
+	</MediaCodec>
 
 
-        <MediaCodec name="c2.intel.mp2.decoder" type="video/mpeg2">
+	<MediaCodec name="c2.intel.mp2.decoder" type="video/mpeg2">
             <Limit name="size" min="64x64" max="2048x2048" />
             <Limit name="alignment" value="2x2" />
             <Limit name="block-size" value="16x16" />
@@ -65,9 +71,9 @@ and updated to vendor media codecs.
             <Limit name="bitrate" range="1-40000000" />
             <Limit name="performance-point-1920x1080" value="30" />
             <Feature name="adaptive-playback" />
-        </MediaCodec>
+	</MediaCodec>
 
-        <MediaCodec name="c2.intel.av1.decoder" type="video/av01">
+	<MediaCodec name="c2.intel.av1.decoder" type="video/av01">
             <Limit name="size" min="64x64" max="4096x4096" />
             <Limit name="alignment" value="2x2" />
             <Limit name="block-size" value="16x16" />
@@ -76,11 +82,11 @@ and updated to vendor media codecs.
             <Limit name="bitrate" range="1-40000000" />
             <Limit name="performance-point-3840x2160" value="30" />
             <Feature name="adaptive-playback" />
-        </MediaCodec>
+	</MediaCodec>
     </Decoders>
 
     <Encoders>
-        <MediaCodec name="c2.intel.avc.encoder" type="video/avc">
+	<MediaCodec name="c2.intel.avc.encoder" type="video/avc">
             <!-- profiles and levels:  ProfileBaseline : Level41 -->
             <Limit name="size" min="176x144" max="4096x4096" />
             <Limit name="alignment" value="2x2" />
@@ -91,16 +97,16 @@ and updated to vendor media codecs.
             <Limit name="performance-point-3840x2160" value="30" />
         </MediaCodec>
 
-        <MediaCodec name="c2.intel.hevc.encoder" type="video/hevc" >
+	<MediaCodec name="c2.intel.hevc.encoder" type="video/hevc" >
             <Limit name="size" min="176x144" max="8192x8192" />
             <Limit name="alignment" value="2x2" />
             <Limit name="block-size" value="16x16" />
             <Limit name="blocks-per-second" range="1-972000" />
             <Limit name="bitrate" range="1-40000000" />
             <Limit name="performance-point-3840x2160" value="30" />
-        </MediaCodec>
+	</MediaCodec>
 
-	    <MediaCodec name="c2.intel.vp9.encoder" type="video/x-vnd.on2.vp9" >
+	<MediaCodec name="c2.intel.vp9.encoder" type="video/x-vnd.on2.vp9" >
             <Limit name="size" min="128x96" max="8192x8192" />
             <Limit name="alignment" value="2x2" />
             <Limit name="block-size" value="16x16" />
@@ -108,6 +114,6 @@ and updated to vendor media codecs.
             <Limit name="blocks-per-second" range="1-972000" />
             <Limit name="bitrate" range="1-40000000" />
             <Limit name="performance-point-3840x2160" value="30" />
-        </MediaCodec>
+	</MediaCodec>
     </Encoders>
 </Included>
diff --git a/caas/extra_files/codecs/media_codecs_gen12.xml b/caas/extra_files/codecs/media_codecs_gen12.xml
index 60b52b4..e6c4bc5 100644
--- a/caas/extra_files/codecs/media_codecs_gen12.xml
+++ b/caas/extra_files/codecs/media_codecs_gen12.xml
@@ -89,9 +89,9 @@ Only the three quirks included above are recognized at this point:
             <Limit name="size" min="64x64" max="4096x4096" />
             <Limit name="alignment" value="2x2" />
             <Limit name="block-size" value="16x16" />
-            <Limit name="block-count" range="1-32768" /> <!-- max 4096x2048 equivalent -->
+            <Limit name="block-count" range="1-65536" /> <!-- max 4096x4096 equivalent -->
             <Limit name="blocks-per-second" range="1-1966080" />
-            <Limit name="bitrate" range="1-48000000" />
+            <Limit name="bitrate" range="1-40000000" />
             <Limit name="performance-point-3840x2160" value="30" />
             <Feature name="adaptive-playback" />
         </MediaCodec>
@@ -101,9 +101,9 @@ Only the three quirks included above are recognized at this point:
             <Limit name="size" min="64x64" max="8192x8192" />
             <Limit name="alignment" value="2x2" />
             <Limit name="block-size" value="8x8" />
-            <Limit name="block-count" range="1-196608" /> <!-- max 4096x3072 -->
-            <Limit name="blocks-per-second" range="1-2000000" />
-            <Limit name="bitrate" range="1-10000000" />
+            <Limit name="block-count" range="1-1048576" /> <!-- max 8192x8192 -->
+            <Limit name="blocks-per-second" range="1-31457280" />
+            <Limit name="bitrate" range="1-40000000" />
             <Limit name="performance-point-3840x2160" value="30" />
             <Feature name="adaptive-playback" />
         </MediaCodec>
@@ -112,8 +112,8 @@ Only the three quirks included above are recognized at this point:
             <Limit name="size" min="64x64" max="8192x8192" />
             <Limit name="alignment" value="2x2" />
             <Limit name="block-size" value="16x16" />
-            <Limit name="block-count" range="1-16384" />
-            <Limit name="blocks-per-second" range="1-500000" />
+            <Limit name="block-count" range="1-262144" />
+            <Limit name="blocks-per-second" range="1-7864320" />
             <Limit name="bitrate" range="1-40000000" />
             <Limit name="performance-point-3840x2160" value="30" />
             <Feature name="adaptive-playback" />
@@ -123,12 +123,12 @@ Only the three quirks included above are recognized at this point:
             <Limit name="size" min="64x64" max="4096x4096" />
             <Limit name="alignment" value="2x2" />
             <Limit name="block-size" value="16x16" />
-            <Limit name="block-count" range="1-16384" />
+            <Limit name="block-count" range="1-131072" />
             <Limit name="blocks-per-second" range="1-500000" />
             <Limit name="bitrate" range="1-40000000" />
             <Limit name="performance-point-3840x2160" value="30" />
             <Feature name="adaptive-playback" />
-        </MediaCodec>
+        </MediaCodec>	
 
         <MediaCodec name="c2.intel.mp2.decoder" type="video/mpeg2">
             <Limit name="size" min="64x64" max="2048x2048" />
@@ -150,12 +150,13 @@ Only the three quirks included above are recognized at this point:
             <Limit name="bitrate" range="1-40000000" />
             <Limit name="performance-point-3840x2160" value="30" />
             <Feature name="adaptive-playback" />
-        </MediaCodec>
+        </MediaCodec>	
     </Decoders>
 
     <Encoders>
 
         <MediaCodec name="c2.intel.avc.encoder" type="video/avc">
+            <!-- profiles and levels:  ProfileBaseline : Level41 -->
             <Limit name="size" min="176x144" max="4096x4096" />
             <Limit name="alignment" value="2x2" />
             <Limit name="block-size" value="16x16" />
-- 
2.39.1

